subcommands/keys/tuf_rotate_all_keys.go

```diff
@@ -0,0 +1,87 @@
+package keys
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/foundriesio/fioctl/subcommands"
+)
+
+func init() {
+	rotate := &cobra.Command{
+		Use:   "rotate-all-keys --keys=<offline-creds.tgz>",
+		Short: "Rotate all online and offline TUF signing keys for the Factory",
+		Long: `Rotate the following TUF keys for the Factory:
+- offline root signing key;
+- offline targets signing key;
+- online targets signing key;
+- online snapshot signing key;
+- online timestamp signing key.
+
+The new signing keys are rotated in both CI and production TUF root transactionally.
+
+When you rotate all TUF signing leys:
+- if there are CI or production targets in your factory, they are re-signed using the new keys.
+- if there is an active wave in your factory, this command is not allowed.
+- new CI targets upload is temporarily disabled for the duration of transaction.`,
+		Example: `
+Migrate an old factory to use Ed25519 key type for all TUF signing keys (online and offline):
+  fioctl keys tuf rotate-all-keys --key-type=ed25519 \
+    --keys=offline-tuf-root-keys.tgz --targets-keys=offline-tuf-targets-keys.tgz`,
+		Run: doRotateAllKeys,
+	}
+	rotate.Flags().StringP("keys", "k", "", "Path to <offline-creds.tgz> used to sign TUF root.")
+	_ = rotate.MarkFlagRequired("keys")
+	_ = rotate.MarkFlagFilename("keys")
+	rotate.Flags().StringP("targets-keys", "K", "", "Path to <offline-targets-creds.tgz> used to sign prod & wave TUF targets.")
+	_ = rotate.MarkFlagFilename("targets-keys")
+	rotate.Flags().BoolP("first-time", "", false, "Used for the first customer rotation. The command will download the initial root key.")
+	rotate.Flags().StringP("key-type", "y", tufKeyTypeNameRSA, "Key type, supported: Ed25519, RSA (default).")
```

Comment:
I wish we could have something like an Enum-type Flag out of the box. So it is easier to write and make it less likely someone will forget to check the valid values down the road in the actual command execution. :slightly_smiling_face:

Reply:
There is this thing: https://pkg.go.dev/github.com/spf13/pflag#Var but I think we could easily go down the rabbit hole with it in Golang, even with the help of generics available in 1.19.